colossalai/builder/pipeline.py

```diff
@@ -250,9 +250,15 @@ def build_pipeline_model(layers: nn.Sequential, num_chunks: int = 1, verbose: bo
     pipeline_parallel_size = gpc.get_world_size(ParallelMode.PIPELINE)
     pipeline_rank = gpc.get_local_rank(ParallelMode.PIPELINE)
     partitions = partition_uniform(len(layers), pipeline_parallel_size, num_chunks)
-    module_list = []
+    layer_marks = []
     for start, end in partitions[pipeline_rank]:
-        module_list.append(nn.Sequential(*layers[start:end]))
+        layer_marks += [i for i in range(start, end)]
+    module_list = []
+    for i, _ in enumerate(layers):
```

Comment:
You can just use `for i in range(layers)`.

Reply:
I am not sure. I need the no. of the layer. `range(layers)` seems to provide the layer name?